subprojects/docs/src/docs/userguide/custom_tasks.adoc

```diff
@@ -141,8 +141,8 @@ If you'd like to optimize your build so that only out-of-date input files are pr
 [NOTE]
 ====
 There is the link:{javadocPath}/org/gradle/api/tasks/incremental/IncrementalTaskInputs.html[IncrementalTaskInputs] API, which is available in Gradle versions before 5.4.
-The usage of this API is discouraged, since it does not allow querying individual input properties for changes.
-The `IncrementalTaskInputs` API will be deprecated and eventually removed.
+The usage of this API is deprecated, and it will be removed eventually.
+The new link:{groovyDslPath}/org.gradle.work.InputChanges.html[InputChanges] API, which is documented here, replaces the old API and addresses its shortcomings.
```

Comment:
We never mention what the shortcomings are. Maybe we should.

Reply:
Should we address the shortcomings in this note in the user manual? Or did you have another place in mind?

Reply:
Since we mention them here I think this is a good place to do it.